pkg/build/api/types.go

```diff
@@ -185,6 +185,11 @@ type CustomBuildStrategy struct {
 	// a validation error is returned.
 	Image string `json:"image"`
 
+	// DockerRegistryAuthRef is used to provide Docker Registry authentication
+	// secret that is used for communication with authentication enabled Docker
+	// Registry. This have to reference existing Secret object.
+	DockerRegistrySecretRef *kapi.ObjectReference `json:"dockerRegistrySecretRef,omitempty"`
```

Comment:
@bparees @soltysh created issue for that, I want to do that as follow up with other properties.

and I like RegistrySecret